ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/AgentUtils.java

```diff
@@ -144,6 +145,20 @@ public class AgentUtils {
     public static final String DEFAULT_DATETIME_PREFIX = "Current date and time: ";
     private static final ZoneId UTC_ZONE = ZoneId.of("UTC");
 
+    public static Map<String, String> extractRequestHeaders(Client client) {
```

Comment:
Are we planning to extract any other headers than `MCP_REQUEST_HEADERS_THREAD_CONTEXT_KEY` in this method if future? If not, suggest to rename the name to be more specific to this use case.